### PR TITLE
[Doc] Standardize versioning and release workflow

### LINKED ISSUE #1198: [RFC] [Workflow] Regular release cycle
Given now we have an increasing amount of users, it's important to release Taichi more frequently and more regularly so that new versions are out in time and in a predictable manner.

## Schedule

We should do two releases per week, one during the middle (~Wednesday) of a week, the other during the weekends.

## Versioning
Since we are still < 1.0.0, we can use minor version bumps for breaking API changes. For example, from 0.6.x to 0.7.0

## Version shepherd

The version shepherd will be in charge of releasing new versions.s

In the past, version releases are mainly done by @yuanming-hu and @archibate. During the GAMES 201, @FantasyVR (Peng Yu) kindly offered his help to release new versions.

TODO: 
 - [ ] @yuanming-hu should document version release workflow and ask @archibate to review.

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = close #1198 

Comments are welcome :-)

Screenshot from 2020-06-11 09-27-22

- [[Click here for the format server]](http://kun.csail.mit.edu:31415/)
- [[Click here for coverage report]](http://codecov.io/gh/taichi-dev/taichi/)

### CODE REVIEW

**Comment by FantasyVR** [comment]
Great, I'll try to release next version at Saturday.

**Comment by archibate** [comment]
Thank for updating this, but I found the [stable](https://github.com/taichi-dev/taichi/tree/stable/) branch is still in v0.6.8 despite v0.6.9 is released, is this intentional?

**Comment by yuanming-hu** [comment]
> Thank for updating this, but I found the [stable](https://github.com/taichi-dev/taichi/tree/stable/) branch is still in v0.6.8 despite v0.6.9 is released, is this intentional?

Just now updated. Sorry, I digressed to write the documentation when I was trying to update `stable`.

File: docs/versioning_releases.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Versioning and releases
```
Comment (archibate) [inline docs/versioning_releases.rst:1]:
```suggestion
Versioning and releaseing
```
Ditto the file name, wdyt?

File: docs/versioning_releases.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+=======================
+
+Pre-1.0 versioning
+------------------
+
+Taichi follows `Semantic Versioning 2.0.0 <https://semver.org/>`_.
+
+Since Taichi is still under version 1.0.0,
+we use minor version bumps for breaking API changes,
+and patch version bumps for backward-compatible changes.
```
Comment (archibate) [inline docs/versioning_releases.rst:11]: Can we given an example on this? e.g. v0.6.10 -> v0.7.0.

File: docs/versioning_releases.rst (hunk 1 above)
Comment (yuanming-hu) [reply docs/versioning_releases.rst:1]: Thanks, but I think "releases" work better here.

**Pull request merged**